src/SDKs/Azure.Base/data-plane/Azure.Base.Tests/TelemetryPolicyTests.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using Azure.Base.Pipeline.Policies;
+using Azure.Base.Testing;
+using NUnit.Framework;
+
+namespace Azure.Base.Tests
+{
+    public class TelemetryPolicyTests: PolicyTestBase
+    {
+        [Test]
+        public async Task ComponentNameAndVersionReadFromAssembly()
+        {
+            var transport = new MockTransport(new MockResponse(200));
+            var telemetryPolicy = new TelemetryPolicy(typeof(TelemetryPolicyTests).Assembly, null);
+
+            var assemblyVersion = Assembly.GetExecutingAssembly().GetName().Version.ToString();
+            await SendGetRequest(transport, telemetryPolicy);
+
+            Assert.True(transport.SingleRequest.TryGetHeader("User-Agent", out var userAgent));
+            Assert.AreEqual(userAgent, $"azsdk-net-base-test/{assemblyVersion} ({RuntimeInformation.FrameworkDescription}; {RuntimeInformation.OSDescription})");
```

Comment:
This is a very specific assert.   Should we consider asserting that each part (version, framework description, etc) is contained in the string somewhere rather than assuming deep knowledge of the exact format?

Reply:
Why do you think it would be more beneficial to avoid asserting the exact format?

Reply:
Generally speaking, the less a test does, the better.   It should have a single responsibility and make assumptions to validate in that area.

The test describes it self as responsible for validating the component name and version are read from the assembly.   Under those semantics, the test is also validating the structure of the user agent string, which is outside its responsibility.    In this case, the format of the UA string can change and still contain the component name and version which should satisfy the test;  to the scope of this test, the UA string format is an implementation detail.

In a wider context, I'd question the usefulness of validating the full structure of the User Agent, as it is an informational element.   So long as it contains the desired information, the format should be meaningless, right?   In the case that the format is, somehow, important, that would warrant another test that validated the format against a pattern, in my opinion.    

Going back to having a single responsibility, I would see the questions "is the string in the correct format"  and "does the string contain this piece of information" to be separate and distinct questions - therefore, separate and distinct tests.

Reply:
I agree with the principles, but I think in this specific case checking everything together is even simpler than trying to assert parts.

You gave me an idea though, I don't think we need user-agent generation code anywhere else so I've made it internal to TelemetryPolicy.